Add documentation for platform dependencies 'provided' scope (#8836)
ballerina-platform/ballerina-dev-website: swan-lake/development-tutorials/source-code-dependencies/package-references.md

**Define the scope for a dependency**

> By default, when the scope has not been explicitly specified for a platform dependency in the `Ballerina.toml`, it will be packaged into the final executable JAR file or the BALA file. There are two scopes that can be used to restrict this behaviour.

Reviewer: Should be `behavior` as per American english. Check all places.

> ***'TestOnly' scope***

Reviewer 2: ```suggestion
***'testOnly' scope***
```

> To restrict a certain platform dependency to be used only for testing, specify the scope as `testOnly`. This will add the platform dependency to the test runtime but will avoid packing it into the final executable JAR file.

Reviewer: ```suggestion
To restrict a certain platform dependency from being used only for testing, specify the scope as `testOnly`. This will add the platform dependency to the test runtime but will avoid packing it into the final executable JAR file.
```

Author: Did not add this change since it gives the idea that the scope will restrict the jar from being used only for testing instead of allowing the jar to be used only for testing, which is the actual purpose of the scope.

> If you want to restrict a certain platform dependency to be used only for testing, specify the scope as `testOnly`. This will add the platform dependency to the test runtime but will avoid packing it into the final executable JAR file.

Reviewer 2: ```suggestion
Tto restrict a certain platform dependency to be used only for testing, specify the scope as `testOnly`. This will add the platform dependency to the test runtime but will avoid packing it into the final executable JAR file.
```

***'provided' scope***

> To restrict a certain platform dependency from been packed into the BALA file, specify the scope as `provided`. This will add the platform dependency to the final executable JAR file but not to the BALA file.

Reviewer: ```suggestion
To restrict a certain platform dependency from being packed into the BALA file, specify the scope as `provided`. This will add the platform dependency to the final executable JAR file but not to the BALA file.
```

> If you want to restrict a certain platform dependency from been packed into the BALA file, specify the scope as `provided`. This will add the platform dependency to the final executable JAR file but not to the BALA file.

Reviewer 2: ```suggestion
To restrict a certain platform dependency from being packed into the BALA file, specify the scope as `provided`. This will add the platform dependency to the final executable JAR file but not to the BALA file.
```

> This scope is useful in cases where the provider's license restricts the redistribution of the platform library. By specifying the "provided" scope, you ensure the dependency is accessible during both compilation and execution, yet it is not included in the BALA. This approach helps avoid any licensing complications associated with redistribution.

Reviewer: consider reword `yet it is not included` , as it is not clear.

> This scope is useful in cases where the provider's license restricts the redistribution of the platform library. By specifying the "provided" scope, you ensure the dependency is accessible during both compilation and execution, yet it isn't included in the BALA package. This approach helps avoid any licensing complications associated with redistribution.

Reviewer 2: ```suggestion
This scope is useful in cases where the provider's license restricts the redistribution of the platform library. By specifying the "provided" scope, you ensure the dependency is accessible during both compilation and execution, yet it is nott included in the BALA. This approach helps avoid any licensing complications associated with redistribution.
```

> In this scenario, when you use such a BALA as a dependency in another project, you'll need to explicitly specify the platform dependency in the Ballerina.toml file since it will not be included in the BALA file itself.

Reviewer 2: ```suggestion
In this scenario, when you use such a BALA as a dependency in another project, you would need to explicitly specify the platform dependency in the Ballerina.toml file since it will not be included in the BALA file itself.
```

Reviewer 2: Let's also mention that provided scope is not supported with build command.

  scope =  "<scope-of-the-jar-file>"
  ```

> When the scope has been specified as `provided`, the values `groupId`, `artifactId` and `version` will be considered mandatory fields for that dependency.

Reviewer 2: Let's make this a note

> >**Note:** When the scope has been specified as `provided`, the values `groupId`, `artifactId` and `version` will be considered mandatory fields for that dependency.

Reviewer: ```suggestion
>**Note:** When the scope has been specified as `provided`, the values `groupId`, `artifactId`, and `version` will be considered mandatory fields for that dependency.
```
